bikesharing/models.py

```diff
@@ -78,11 +78,19 @@ def __repr__(self):
         return "#{bike_number} ({state}) @ {position} ({last_reported})".format(
             bike_number=self.bike_number,
             state=self.state,
-            position=self.current_position(),
+            position=self.public_geolocation(),
             last_reported=self.last_reported,
         )
 
-    def current_position(self):
+    def public_geolocation(self):
+        if not self.id:
+            return None
+        try:
+            return Location.objects.filter(bike=self, internal_location=False).latest('reported_at')
+        except ObjectDoesNotExist:
+            return None
+
+    def internal_geolocation(self):
```

Comment:
internal geolocation also returns locations that are not internal. is this on purpose?

Reply:
idea was to have definitiv the latest location. but open for discussion. In Bike i only display internal location, if it differs from the public location

Reply:
changed it now. I think its now less confusing for users